File: 0070/70.climbing-stairs.java
/*
 * @lc app=leetcode id=70 lang=java
 *
 * [70] Climbing Stairs
 */

// @lc code=start
class Solution {
    public int climbStairs(int n) {

        if (n == 0) {
            return 0;
        }
        
        if (n == 1){
            return 1;
        }

        int first = 0;
        int second = 1;
        int nth = 0;

        for (int i = 1; i <= n; i++) {
            nth = first + second;
            first = second;
            second = nth;
        }
        return nth;

        
    }

}
// @lc code=end

File: 0070/70.climbing-stairs.py
#
# @lc app=leetcode id=70 lang=python3
#
# [70] Climbing Stairs
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 0:
            return 0
        
        if n == 1:
            return 1
        
        first = 0
        second = 1
        nth = 0

        for x in range(n):
            nth = first + second
            first = second
            second = nth
        
        return nth
# @lc code=end

